### PR TITLE
Run e2e tests on GH actions

### PR DESCRIPTION
**Pull request by nicolo-ribaudo**
| Q                        | A 
| ------------------------ | ---
| License                  | MIT

CircleCI is quite slow because there is a low concurrency limit (I think it's 1 job at time?) so it gets overloaded easily and sometimes jobs are queued for hours. Even when they are not, it happens quite often that they wait for ~30 mins before running.

This PR moves the Babel 7 e2e tests to GitHub actions, where the concurrency limit is 16 parallel jobs, to speed it up a bit.

I'm not moving _everything_ to GH actions to avoid hitting the concurrency limit. CircleCI will still run:
- `build-standalone` on PRs and `main`
- `e2e-breaking` on `main` (it's opt-in on PRs)
- `test262` on `main` (it's opt-in on PRs)

Thanks @juanpicado for the help!

<a href="https://gitpod.io/#https://github.com/babel/babel/pull/12517"></a>

### CODE REVIEW

**Comment by juanpicado** [comment]
You mentioned you were using the httpassw file, I'm not sure whether you wants to use it, I'd use for CI https://github.com/verdaccio/monorepo/tree/9.x/plugins/auth-memory#verdaccio-auth-memory which does not require write anything also speed up things a bit.

**Comment by nicolo-ribaudo** [comment]
Thanks, I'm trying that now.

**Comment by juanpicado** [comment]
I'll take a look over this weekend to this, let's see whether I find something.

**Comment by juanpicado** [comment]
Ok fixed :+1: , I've spotted few little mistakes, do you mind if I PR on top of this one? I'd like to apply few optimizations. 

Screenshot from 2021-01-09 19-08-48

**Comment by juanpicado** [comment]
Changes are here https://github.com/juanpicado/babel/pull/1/files

**Comment by nicolo-ribaudo** [comment]
Hey sorry I completely forgot about this. I'll merge your branch into mine!

**Comment by juanpicado** [comment]
Sure, just let me know if you need some help, I'll be willing to move this on.

File: scripts/integration-tests/utils/local-registry.sh [hunk 1]
```diff
@@ -3,14 +3,14 @@
 # Copied from https://github.com/facebook/create-react-app/blob/053f9774d3f592c17741d2a86de66a7ca58f90c0/tasks/local-registry.sh
 
 custom_registry_url=http://localhost:4873
-default_verdaccio_package=verdaccio@~4.3.3
+default_verdaccio_package=verdaccio@~4.10.0
```
Comment (juanpicado) [inline scripts/integration-tests/utils/local-registry.sh:6]: I think is better `v4.11.1`, it has few optimizations.

File: scripts/integration-tests/utils/local-registry.sh [hunk 2]
```diff
@@ -3,14 +3,14 @@
...
 
 function startLocalRegistry {
   # Start local registry
   tmp_registry_log=`mktemp`
   echo "Registry output file: $tmp_registry_log"
   (cd && nohup npx ${VERDACCIO_PACKAGE:-$default_verdaccio_package} -c $1 &>$tmp_registry_log &)
-
+  npm install --global verdaccio-memory
```
Comment (juanpicado) [inline scripts/integration-tests/utils/local-registry.sh:13]: I'd pin `verdaccio-memory@9.7.2` . There are few major coming soon that might affect the pipeline.

File: scripts/integration-tests/verdaccio-config.yml [hunk 3]
```diff
@@ -20,6 +20,10 @@ auth:
 uplinks:
   npmjs:
     url: https://registry.npmjs.org/
+    agent_options:
```
Comment (juanpicado) [inline scripts/integration-tests/verdaccio-config.yml:23]: The `agent_options` are not need it if you use (v4.10.0). It is harmless if you like to keep it.

File: scripts/integration-tests/verdaccio-config.yml [hunk 4]
```diff
@@ -54,7 +58,7 @@ middlewares:
 
 # log settings
 logs:
-  - { type: stdout, format: pretty, level: http }
+  - { type: stdout, format: pretty, level: trace }
```
Comment (juanpicado) [inline scripts/integration-tests/verdaccio-config.yml:61]: I'd move here to `warn` to speed boost. `trace` is really expensive.

**Review by juanpicado — APPROVED** [review]
LGTM

File: scripts/integration-tests/publish-local.sh [hunk 5]
```diff
@@ -31,7 +31,7 @@ loginLocalRegistry
 # This script gets the last @babel/standalone version (because it's always published),
```
Comment (JLHwung) [inline scripts/integration-tests/publish-local.sh:31]:
```suggestion
# This script gets the last root package.json version,
```

File: scripts/integration-tests/utils/local-registry.sh [hunk 6]
```diff
@@ -3,14 +3,14 @@
...
-default_verdaccio_package=verdaccio@~4.3.3
+default_verdaccio_package=verdaccio@~4.11.1
 
 function startLocalRegistry {
   # Start local registry
   tmp_registry_log=`mktemp`
   echo "Registry output file: $tmp_registry_log"
   (cd && nohup npx ${VERDACCIO_PACKAGE:-$default_verdaccio_package} -c $1 &>$tmp_registry_log &)
-
+  npm install --global verdaccio-memory@~9.7.2
```
Comment (JLHwung) [inline scripts/integration-tests/utils/local-registry.sh:13]: Can it be installed from `yarn`? The yarn cache should helps here.

**Review by JLHwung — APPROVED** [review]
LGTM with some nits.

File: scripts/integration-tests/utils/local-registry.sh (hunk 6 above)
Comment (nicolo-ribaudo) [reply scripts/integration-tests/utils/local-registry.sh:13]:
```suggestion
  verdaccio-memory@~9.7.2 verdaccio-memory@~9.7.2
```

**Comment by nicolo-ribaudo** [comment]
I'm merging even if one review is from @juanpicado and not from a Babel team member, since he definitely has more experience than us with Verdaccio on CI servers :stuck_out_tongue:

**Pull request merged**